File: 541. Reverse String II/solution.java
class Solution {
  public String reverseStr(String s, int k) {
    String result = "";
    Boolean reversed = true;
    int start = 0;
    int end = k;
    while (start < s.length()){
      if (end > s.length()){
        end = s.length();
      }
      if (reversed){
        String newString = s.substring(start, end);
        String resultingString = "";
        for (int index = newString.length() - 1; index >  -1; index--){
          resultingString += newString.charAt(index);
        }
        result += resultingString;
      }
      else{
        result += s.substring(start, end);
      }
      reversed = !reversed;
      start += k;
      end += k;
    }

    return result;
  }
}

File: 541. Reverse String II/solution.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
 var reverseStr = function(s, k) {
  let result = "";
  let reversed = true;
  let start = 0;
  let end = k;
  while (start < s.length){
    if (reversed){
      result += s.slice(start, end).split("").reverse().join("");
    }
    else{
      result += s.slice(start, end);
    }
    reversed = !reversed;
    start += k;
    end += k;
  }

  return result;
};